src/routes/+layout.svelte

```diff
@@ -28,16 +28,20 @@
 				// Unauthenticated user wants to go to a protected route
 				goto('/login');
 				return;
-			}
-			if (!user && noAuthRoutes.includes(currentPath)) {
+			} else if (!user && noAuthRoutes.includes(currentPath)) {
 				// Unauthenticated user wants to go to a public route
 				goto(currentPath);
 				return;
-			}
-			if (user) {
-				// Authenticated user
+			} else if (user && noAuthRoutes.includes(currentPath)) {
+				// Authenticated user wants to go to a public route
+				goto('/dashboard');
+				return;
+			} else if (user && !noAuthRoutes.includes(currentPath)) {
+				// Authenticated user wants to go to a public route
 				goto(currentPath);
 				return;
+			} else {
+				return;
```

Comment:
This block can be removed altogether, right?

Reply:
Yeah, you're right. I wanted to add a default return. Maybe it's not necessary, or at least doesn't have to be in an else block. Removed it for now